src/App.jsx

```diff
@@ -5,15 +5,29 @@ import AddBlogTitle from './AddBlogTitle.jsx'
 import Search from './Search.jsx'
 
 function App() {
-    const [newTitle, setNewTitle] = react.useState();
+    const initialTitles = ["Lesson 1.1 Project setup",
+        "Lesson 1.2 React DOM and components",
+        "Lesson 1.3 Form input, props, state and callbacks",
+        "Lesson 1.4 Lifting state, controlled components, props handling",
+        "Lesson 1.5 Side effects, custom hooks, fragments",
+    ]
+    const blogTitlesFromLocalStorage = JSON.parse(localStorage.getItem('savedBlogTitles')) || initialTitles
+    const [blogTitles, setBlogTitles] = react.useState(blogTitlesFromLocalStorage);
     const [searchText, setSearchText] = react.useState('');
+
+    // run side effect when blogTitles state changes and save to localstorage
+    react.useEffect(() => {
+        localStorage.setItem('savedBlogTitles', JSON.stringify(blogTitles))
+    }, [blogTitles])
+
     return (
-        <div className="App">
+        <div>
             <h1>Cockatoo Lesson Blog Title</h1>
-            <AddBlogTitle setNewTitle={setNewTitle}/>
+            <AddBlogTitle onAddTitle={(newTitle) => {
+                setBlogTitles([...blogTitles, newTitle]) // react does state update in background, asynchronously
```

Comment:
We tried saving `blogTitles` to localstorage in this callback function after setting the blog titles but found that `blogTitles` did not have the `newTitle` added to the list! This is because react does state updates in the background asynchronously. React had not yet finished updating `blogTitles` with the extra title by the time we tried saving to localstorage.